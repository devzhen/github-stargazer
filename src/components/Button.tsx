import React from 'react';
import { ActivityIndicator } from 'react-native';
import styled from 'styled-components/native';

export enum ButtonBackground {
  Orange = 'orange',
  Gray = 'gray',
}

interface ButtonProps {
  labelSize: number;
  label: string;
  bgColor: ButtonBackground;
  onPress: any;
  disabled: boolean;
  loading?: boolean;
}

const Button = ({
  labelSize,
  label,
  bgColor,
  onPress,
  disabled,
  loading,
}: ButtonProps) => (
  <Wrapper
    onPress={() => {
      if (disabled || loading) {
        return;
      }
      onPress();
    }}
    bgColor={bgColor}
    disabled={disabled}
  >
    {!loading && <Label labelSize={labelSize}>{label}</Label>}
    {loading && <ActivityIndicator size="small" color="#fff" />}
  </Wrapper>
);

Button.defaultProps = {
  onPress: () => {},
  labelSize: 17,
  label: 'No data',
  bgColor: ButtonBackground.Orange,
  disabled: false,
  loading: false,
};

const Wrapper = styled.TouchableOpacity<{ bgColor: string; disabled: boolean }>`
  flex: 1;
  height: 100%;
  align-items: center;
  flex-direction: row;
  justify-content: center;
  ${(props) =>
    props.bgColor === ButtonBackground.Orange &&
    `background-color: ${props.theme.colors.orange}`}
  ${(props) =>
    props.bgColor === ButtonBackground.Gray &&
    `background-color: ${props.theme.colors.gray}`}
  border-radius: 4px;
  ${(props) => props.disabled && 'opacity: 0.7'};
`;

const Label = styled.Text<{ labelSize: number }>`
  font-size: ${(props) => props.labelSize}px;
  text-align: center;
  color: #fff;
`;

export default Button;
