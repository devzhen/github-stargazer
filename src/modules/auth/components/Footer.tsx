import React from 'react';
import { View, StyleSheet } from 'react-native';

import { IS_IPHONE_X } from '@helpers/deviceInfoHelper';

interface FooterProps {
  children: React.ReactNode;
}

const Footer = ({ children }: FooterProps) => (
  <View style={styles.container}>{children}</View>
);

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: 50,
    paddingHorizontal: 46,
    marginBottom: IS_IPHONE_X ? 100 : 80,
    marginTop: 20,
  },
});

export default Footer;
