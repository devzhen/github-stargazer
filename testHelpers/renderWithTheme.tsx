import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import 'jest-styled-components';

import theme from '@theme/index';

const renderWithTheme = (component: any) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export default renderWithTheme;
