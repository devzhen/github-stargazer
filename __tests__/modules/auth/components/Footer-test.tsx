import React from 'react';
import 'jest-styled-components';

import Button from '@components/Button';

import Footer from '@modules/auth/components/Footer';

// @ts-ignore
import renderWithTheme from '@testHelpers/renderWithTheme';

describe('Footer auth component', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Footer>
        <Button label="Login" onPress={() => {}} loading={true} />
      </Footer>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
