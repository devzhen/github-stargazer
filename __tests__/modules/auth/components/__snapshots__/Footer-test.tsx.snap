// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer auth component renders correctly 1`] = `
<View
  style={
    Object {
      "height": 50,
      "marginBottom": 80,
      "marginTop": 20,
      "paddingHorizontal": 46,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#eb9e38",
        "borderRadius": 4,
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#fff"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
</View>
`;
